extensions/resource-deployment/src/ui/notebookWizard/notebookWizard.ts

```diff
@@ -57,9 +67,13 @@ export class NotebookWizard extends WizardBase<NotebookWizard, NotebookWizardPag
 		await this.openNotebook(notebook);
 	}
 	protected async onOk(): Promise<void> {
-		const notebook = await this.prepareNotebookAndEnvironment();
-		const openedNotebook = await this.openNotebook(notebook);
-		openedNotebook.runAllCells();
+		if (instanceOfNotebookWizardDeploymentProvider(this.resourceProvider)) {
```

Comment:
I have this working assumption: A particular wizard is only intended to work for a given provider, if that is indeed correct then we should ensure that no part of the wizard can ever get called with any other provider instance so that such checks are unnecessary.